### PR TITLE
Add content to the Readme

### PR DESCRIPTION
**Pull request by legionth**
The `README.md` needed some ❤️ 

What did I do in this PR?
* Adpated the description
* Added a table of contents
* Added some new chapters with code snippets frpm the `example.php`
* Added links to external sources like the official API documentation and php.net

Feel free to give me feedback on this 👍

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,150 @@
...
+		'status' => [
+			['op' => '=', 'val' => 1],
+		],
+	],
+];
+```
+
+### Request
+
+To create a request to the API a an `ACTION_ID` is needed.
```
Comment (dkaisers) [inline README.md]:
```suggestion
To create a request to the API an `ACTION_ID` is needed.
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,150 @@
...
+In the class `onOfficeSDK` are several constants defined,
```
Comment (dkaisers) [inline README.md]:
```suggestion
The class `onOfficeSDK` defines several constants,
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,150 @@
...
+that can be used, so there is no need to copy these `ACTION_IDs`. 
+
+To send the request an API token and API secret is needed,
+checkout the [official API documentation](#api-documentation)
+how to get these credentials.
```
Comment (dkaisers) [inline README.md]: I don't think that this sentence is quite correct. Maybe use something more like this?

> A **token** and **secret** are needed to send a request to the API. Check out the [official API documentation](#api-documentation) for information on how to aquire these credentials.

File: README.md [hunk 4]
```diff
@@ -1,23 +1,150 @@
...
+The return value of `onOfficeSDK::callGeneric` is used to identify the
+equivalent response value.
+`onOfficeSDK::callGeneric` can be called multiple times before sending
+the request to the API via `onOfficeSDK::sendRequests`-
+
+
+### Repsonse
+
+Use the method `onOfficeSDK::getResponseArray` to fetch the data for the response data.
+To identify the response of the request use the return value `onOfficeSDK::callGeneric`.
```
Comment (dkaisers) [inline README.md]:
```suggestion
To identify the response of the request use the value returned by `onOfficeSDK::callGeneric`.
```

File: README.md [hunk 5]
```diff
@@ -1,23 +1,150 @@
...
+
+```php
+var_export($pSDK->getResponseArray($handleReadEstate));
+```
+
+The response will be an array values.
```
Comment (dkaisers) [inline README.md]:
```suggestion
The response will be a PHP array.
```

File: README.md [hunk 6]
```diff
@@ -1,23 +1,150 @@
...
+
+## API Documentation
+
+The API client is developed for the current version of the official API version.
```
Comment (dkaisers) [inline README.md]: Maybe use *latest* instead of *current* as this is also the identifier used by the SDK.

**Comment by legionth** [comment]
@dkaisers thanks for the review!
I added your feedback in the latest commit :)

File: README.md [hunk 7]
```diff
@@ -1,23 +1,149 @@
...
+```php
+$handleReadEstate = $pSDK->callGeneric(onOfficeSDK::ACTION_ID_READ, 'estate', $parametersReadEstate);
+
+$pSDK->sendRequests('put the token here', 'and secret here');
+```
+
+The return value of `onOfficeSDK::callGeneric` is used to identify the
+equivalent response value.
+`onOfficeSDK::callGeneric` can be called multiple times before sending
+the request to the API via `onOfficeSDK::sendRequests`-
```
Comment (dkaisers) [inline README.md]:
```suggestion
the request to the API via `onOfficeSDK::sendRequests`.
```

File: README.md [hunk 8]
```diff
@@ -1,23 +1,149 @@
...
+This project is the official PHP API client for the
+[onOffice API](https://apidoc.onoffice.de/)
+supported by the onOffice GmbH.
 
-It contains code for the proper generation of a request with a given token, secret, request action
-and parameters.
-There is no information about API actions and their proper parameters included in
-this package.
+* **The HTTP protocol** is used to communicate with the API.
+* An **Access Token** is used to ensure a **secure** communication with the API.
```
[the remark this answers is not in the document]
Comment (legionth) [reply README.md]: Will change it to `An **Access Token** and a **TLS connection** are used to ensure a **secure** communication with the API.`

**Comment by jayay** [comment]
LGTM 👍

**Pull request merged**